src/shogun/mathematics/graph/nodes/Exp_test.cc

```diff
@@ -0,0 +1,50 @@
+#include <gtest/gtest.h>
+
+#include <shogun/mathematics/graph/Graph.h>
+#include <shogun/mathematics/graph/nodes/Input.h>
+#include <shogun/mathematics/graph/nodes/Exp.h>
+
+#include "../test/GraphTest.h"
+
+using namespace shogun;
+using namespace shogun::graph;
+using namespace std;
+
+TYPED_TEST(GraphTest, exp)
+{
+	using NumericType = typename TypeParam::c_type;
+
+	if constexpr (!std::is_same_v<NumericType, bool>) 
+	{
+		SGVector<NumericType> X{0, 1, 10};
+
+		SGVector<NumericType> expected_result(X.size());
+
+		std::transform(X.begin(), X.end(), expected_result.begin(), 
+			[](const NumericType& el) {return std::exp(el);});
+
+		auto input = make_shared<node::Input>(
+		    Shape{Shape::Dynamic}, TypeParam::type_id);
+
+		auto output = make_shared<node::Exp>(input);
+
+		auto graph = make_shared<Graph>(
+		    vector{input}, vector<shared_ptr<node::Node>>{output});
+
+		for (auto&& backend : this->m_backends)
```

Comment:
question: is it good that tests for ops need to have all this stuff in them? I.e. I think it should be the same roughly for other ops, so some code could be reused?

Reply:
@karlnapf I just wanted to be explicit with tests for now, otherwise I get myself lost because it's changing so often.. Once the API is more stable and logical, and the code works fine, I'll refactor the tests :)

Reply:
also, I am working on the Array version now, which basically brings the functional version, and then need to combine that to the tests too

Reply:
makes sense!